.github/workflows/deploy-to-production.yml

```diff
@@ -0,0 +1,169 @@
+name: Deploy to AWS Production
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        description: 'Part of version to increment (major, minor, patch)'
+        required: true
+        default: 'patch'
+        type: choice
+        options: [patch, minor, major]
+      release_title:
+        description: 'Optional title to replace auto-generated one'
+        required: false
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests:
+    uses: ./.github/workflows/ci.yml
+  notify:
+    runs-on: ubuntu-latest
+    needs: ["tests"]
+    steps:
+      - run: echo 'Linter and tests passed.'
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        fetch-depth: 0  # fetch full history and tags
+
+      - name: Get latest tag
+        id: get_tag
+        run: |
+          latest=$(git describe --tags --abbrev=0 || echo "v0.0.0")
+          echo "Latest tag is $latest"
+          echo "tag=$latest" >> "$GITHUB_OUTPUT"
+
+      - name: Bump version
+        id: bump
+        run: |
+          version="${{ steps.get_tag.outputs.tag }}"
+          version="${version#v}"
+          IFS='.' read -r major minor patch <<< "$version"
+
+          case "${{ github.event.inputs.release_type }}" in
+            major)
+              major=$((major + 1)); minor=0; patch=0;;
+            minor)
+              minor=$((minor + 1)); patch=0;;
+            patch)
+              patch=$((patch + 1));;
+            *)
+              echo "Invalid release_type"; exit 1;;
+          esac
+
+          new_version="v$major.$minor.$patch"
+          echo "new_tag=$new_version" >> "$GITHUB_OUTPUT"
+          echo "Next tag: $new_version"
+
+      - name: Create and push new tag
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git tag ${{ steps.bump.outputs.new_tag }}
+          git push origin ${{ steps.bump.outputs.new_tag }}
+
+      - name: Set up GitHub CLI
+        uses: cli/gh-action@v2
+
+      - name: Create GitHub Release with auto-generated notes
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          TITLE="${{ github.event.inputs.release_title }}"
+          if [ -z "$TITLE" ]; then
+            TITLE="${{ steps.bump.outputs.new_tag }}"
+          fi
+          echo "Using release title: $TITLE"
+          gh release create ${{ steps.bump.outputs.new_tag }} --title "$TITLE" --generate-notes
```

Comment:
this is gh autogenerated, so not a lot of room for us to edit it, i think there is an option to pass in notes that are edited, but just wanted to start with this version b/c it's out of the box & i dont want to waste too much time on making this perfect yet